_posts/2022-12-12-bentoml-deployment-with-zenml.md

```diff
@@ -0,0 +1,269 @@
+
+```python
+from zenml.integrations.bentoml.steps import (
+    BentoMLBuilderParameters,
+    bento_builder_step,
+)
+
+bento_builder = bento_builder_step(
+    params=BentoMLBuilderParameters(
+        model_name=MODEL_NAME,
+        model_type="pytorch",
+        service="service.py:svc",
+        labels={
+            "framework": "pytorch",
+            "dataset": "mnist",
+            "zenml_version": "0.21.1",
+        },
+        exclude=["data"],
+        python={
+            "packages": ["zenml", "torch", "torchvision"],
+        },
+    )
+)
+```
+
+This step takes the trained model and creates a Bento artifact that can be used for deployment. This is an important step because it allows us to easily share and deploy our trained model to different environments.
+
+The last step in our pipeline is the deployment step, which deploys the packaged file called the Bento that contains everything we need to deploy the trained model. The deployment step is also a built-in ZenML step that can be easily referenced to deploy the model to a local machine:
+
+```python
+from zenml.integrations.bentoml.steps import (
+    BentoMLDeployerParameters,
+    bentoml_model_deployer_step,
+)
+
+bentoml_model_deployer = bentoml_model_deployer_step(
+    params=BentoMLDeployerParameters(
+        model_name=MODEL_NAME,
+        port=3001,
+        production=False,
+    )
+)
+```
+After defining all these steps and the pipeline we can run the pipeline which will give us the following:
+
+![Pipeline terminal logs]({{ site.url }}/assets/posts/bentoml-deployment/pipeline-terminal-logs.png)
+
+In the ZenML dashboard we can see the pipeline dag:
+
+![Pipeline dag]({{ site.url }}/assets/posts/bentoml-deployment/pipeline-dag.png)
+
+In the Swagger UI, we can see more details about the deployed bento:
+
+![Swagger UI]({{ site.url }}/assets/posts/bentoml-deployment/swagger-ui.png)
+
+## Inference Pipeline:
+
+To make use of our deployed model the example comes with an inference pipeline which is a set of steps that use some image samples to run prediction on it. You can find the code of this pipeline and more in the Bentoml_Deployment example.
+
+![Inference pipeline]({{ site.url }}/assets/posts/bentoml-deployment/inference-pipeline.png)
+
+## From Local to Cloud:
+
+Once a model server has been deployed and tested locally, it can be deployed to a cloud provider using Bentoctl, a command-line interface (CLI) tool for managing BentoML deployments on cloud providers. Bentoctl currently supports AWS, GCP, and Azure, and requires that you have an account with one of these providers as well as the corresponding CLI tools and Docker and Terraform installed on your machine.
```

Comment:
```suggestion
Once a model server has been deployed and tested locally, it can be deployed to a cloud provider using `bentoctl`, a command-line interface (CLI) tool for managing BentoML deployments on cloud providers. Bentoctl currently supports AWS, GCP, and Azure, and requires that you have an account with one of these providers as well as the corresponding CLI tools and Docker and Terraform installed on your machine.
```